README
## Copyright (c) 2012, Jake "redmaner" van der Putten
## Engengis project
## Version: Delta 
## Universal build 46
-------------------------------------------------------------------------------------
Conditions off the source
-------------------------------------------------------------------------------------
You may use the source to build engengis yourself including your own code and improvements.
You may use the source to build engengis specially for your ROM
You may use the source to contribute to engengis project and report bugs or improvements.
You may NOT use the source to build engengis and release it under another name!
You may NOT use the source to include it in another script package!
If you have any questions about the source and using it contact me jakevdputten@live.nl (it's my second email adress)

--------------------------------------------------------------------------------------
Engengis official repo
--------------------------------------------------------------------------------------
Includes:
- Latest source code
- OTA update service source
- Old scripts
- Build-tool 

--------------------------------------------------------------------------------------
How to build engengis?
--------------------------------------------------------------------------------------
To build engengis you need a few things:
- Linux machine (ubuntu preffered)(not tested on Mac OSX)(Not working on windows)
- Java installed (to sign zip)
- zip/unzip (to make zip)(to get it: apt-get install zip)(apt-get install unzip)

Browse to engengis repo on your machine and give "build-engengis.sh" executable permissions.
Run script and let it do it's thing. 
On some devices you have to edit "build-engengis.sh" to let it work on your machine.

--------------------------------------------------------------------------------------
What is engengis?
--------------------------------------------------------------------------------------
Engengis is like many others a script for android devices.
There is only a huge diffrence between Engengis and other scripts.
Most scripts are totally based on improving something like better RAM or battery.
Engengis is doing something else: it's 100% configureable by the end user.

The main goal is that the end-user desides what's the best for his device.
You must admit all devices are diffrent same as the user off it! So wye should I decide what's best for him?
To achieve the best result Engengis has a lot off features that are configureable.
Not only the features are important but also the knowledge off the user.
That's wye we offer some detailed information about the features we offer.

--------------------------------------------------------------------------------------
Features
--------------------------------------------------------------------------------------
- Easy to use interface
- Secure to use (can't run multiply tweaks at once)
- All features can be enabled/disabled
- Supports most android devices
- Checks the requirements that are needed to use engengis
- Supports a lot off partition types: BML - MMC - MTD - STL - TFSR - ZRAM
- Offers a lot off tweaks/scripts
- Has it's own OTA updater
- Engengis menu to configure engengis
- Remembers your tweaks when updated so you run always the most up-to-date scripts after an update
- Works on most android versions

--------------------------------------------------------------------------------------
Included scripts/tweaks
--------------------------------------------------------------------------------------
- Engengis configureable Systemtweak
- Zipalign during boot
- optimize sqlite db's (not as boot but using the interface)
- drop caches during boot
- Internetspeed tweaks 
- Internetsecurity tweaks
- IO schedulers BFQ - CFQ - Deadline - Noop - SIO - VR (can't run multiply schedulers with engengis)
- check wich I/O schedulers are supported
- SD-Readspeeds 256 - 512 - 1024 -2048 - 3072 - 4096
- CPU governors (can only be set when kernel supports them)
- Conservative - ondemand - smartassV2 - interactive - lulzactive  supported
- Possibility to set display resolution (dpi)
- Has a few build.prop tweaks included

See: http://code.google.com/p/engengis

--------------------------------------------------------------------------------------
Credits/maintainers
--------------------------------------------------------------------------------------
Engengis has been made using some other existing code and by studying existing code.
The authors of that code are stated below. Without them Engengis wasn't the way it is now.

- Brainmaster [SD-Readspeed scripts]
- GingerReal community [Helped with testing + debugging]
- Google [Used LMK settings from ICS and GB]
- Hardcore [Noatime mount using loop]
- KNZO [void menu][IO scheduler + cpu governor information]
- Linux [Studied source code documentation]
- Pikachu01 [Optimiz sqlite db's script][Dropcaches idea]




CHANGELOG
-------------------------------------------------------------------------------------------------------------
Engengis.Delta_build-46
-------------------------------------------------------------------------------------------------------------
- Initial rootzwiki release
- Reorganized main code
- Added version info feature
- Added configurable systemtweak
- Added support for MTD blocks
- Added CPU governor tweaks for conservative and ondemand
- Added force update option in OTA service
- Added internetspeed tweaks
- Added internetsecurirty tweaks
- Improved visaul appearance
- Improved "Uninstall" feature
- Improved recommanded settings
- Improved restore settings
- Fixed busybox bug
- Fixed log bug 
- Removed performance governor

-------------------------------------------------------------------------------------------------------------
Engengis.Charlie_build-44
-------------------------------------------------------------------------------------------------------------
- Optimized the whole code (it's a lot faster now)
- Merged all menu's in one file (it's a lot better and faster)
- Added save feature (remebers your tweaks now)
- Added configuration file
- Added new requirements check
- Added log for requirments only
- Added engengis settings to configure engengis
- Added a disclaimer
- Added support for MTD and ZRAM blocks
- Added optimize ramleaks during engengis execution
- Added build.prop optimizations menu
- Added support for performance and interactive governor
- Improved HSS tweak (two version now)
- Improved DPI menu
- Improved disable all tweaks feature - Improved OTA update service - Improved a lot off other features
- Removed lionhearth governor support
- Removed advanced features menu
- Removed fugu tweaks + restore option
- Removed dalvik cache verification tweaks

-------------------------------------------------------------------------------------------------------------
Engengis.Bravo_build-28
-------------------------------------------------------------------------------------------------------------
- Initial public release to gingerreal forums
- Added OTA support
- Added dropcaches script
- Added sqlite optimize option
- Added advanced features menu
- Added dpi menu
- Improved HSS tweak - Improved security
- Improved all IO schedulers
- Improved zipalign script
- Improved some governors
- Increased menu speed
- Fixed a lot off bugs

-------------------------------------------------------------------------------------------------------------
Engengis.Alpha_build-1
-------------------------------------------------------------------------------------------------------------
- Initial test team release


installer.sh
#!/system/bin/sh
# Copyright© 2011 redmaner
# Engengis installer V6

clear
echo
echo " ------------------------"
echo "    Engengis.Installer   " 
echo " ------------------------"
echo 
wget -q http://dl.dropbox.com/u/26139869/engengis/currentversion -O /sdcard/currentversion
echo "Installer will install: `cat /sdcard/currentversion`"
sleep 2
echo
echo "Do you want to proceed?"
echo "[ y or n]"
read option

case "$option" in 
 
  mount -o remount,rw /system
  echo
  echo "This will take a moment..."
  mount -o remount,rw /system
  rm -rf /system/etc/engengis
  rm -f /system/bin/engengis
  rm -f /system/bin/engengisa
  rm -f /system/bin/engengissettings
  rm -f /system/bin/engengisg
  rm -f /system/bin/engengishss
  rm -f /system/bin/engengisr
  rm -f /system/bin/engengiss
  rm -f /system/xbin/sqlite3
  rm -f /system/xbin/zipalign
  rm -f /system/etc/init.d/S00ramscript
  rm -f /system/etc/init.d/S07hsstweak
  rm -f /system/etc/init.d/S14zipalign
  rm -f /system/etc/init.d/S21sqlite
  rm -f /system/etc/init.d/S28scheduler
  rm -f /system/etc/init.d/S35sdreadspeed
  rm -f /system/etc/init.d/S42cpugovernor
  rm -f /system/etc/init.d/S49dropcaches
  rm -f /system/etc/init.d/S56internet
  rm -f /system/etc/init.d/S63internetsecurity
  rm -f /data/engengis.log
  echo "Installing new files..."
  mkdir /system/etc/engengis
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/bin/engengis -O /system/bin/engengis
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S07hsstweak -O /system/etc/engengis/S00ramscript
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S14zipalign -O /system/etc/engengis/S14zipalign
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S21hsstweak -O /system/etc/engengis/S21hsstweak
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28bfqscheduler -O /system/etc/engengis/S28bfqscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28cfqscheduler -O /system/etc/engengis/S28cfqscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28deadlinescheduler -O /system/etc/engengis/S28deadlinescheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28noopscheduler -O /system/etc/engengis/S28noopscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28sioscheduler -O /system/etc/engengis/S28sioscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28vrscheduler -O /system/etc/engengis/S28vrscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd256 -O /system/etc/engengis/S35sd256
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd512 -O /system/etc/engengis/S35sd512
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd1024 -O /system/etc/engengis/S35sd1024
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd2048 -O /system/etc/engengis/S35sd2048
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd3072 -O /system/etc/engengis/S35sd3072
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd4096 -O /system/etc/engengis/S35sd4096
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpuconservative -O /system/etc/engengis/S42cpuconservative
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpululzactive -O /system/etc/engengis/S42cpululzactive
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpuinteractive -O /system/etc/engengis/S42cpuinteractive
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpuondemand -O /system/etc/engengis/S42cpuondemand
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpusmartassv2 -O /system/etc/engengis/S42cpusmartassv2
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S49dropcaches -O /system/etc/engengis/S49dropcaches
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S56internet -O /system/etc/engengis/S56internet
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S63internetsecurity -O /system/etc/engengis/S63internetsecurity
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/version -O /system/etc/engengis/version
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/xbin/sqlite3 -O /system/xbin/sqlite3
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/xbin/zipalign -O /system/xbin/zipalign
  echo "Setting permissions..."
  chmod 777 /system/bin/engengis
  chmod 777 /system/xbin/sqlite3
  chmod 777 /system/xbin/zipalign
  if [ -e /data/settings.conf ]; then
      sed -i '/status=*/ d' /data/engengis.conf;
      echo "status=updated" >> /data/engengis.conf;
  fi;
  echo
  echo "Installation complete!"
  sleep 5
  clear 
  engengis
  ;;
  "n" | "N")
  clear
  exit
  ;;
esac


build-engengis.sh
#!/bin/sh
# Copyright (c) 2012, Jake "redmaner" van der Putten
# Engengis build tool
# Version: 21012012

# System requirements
# -----------------------------------------
# - Ubuntu (or any other unix based system)
# - Java (to sign zip file)
# - zip/unzip (apt-get install zip/apt-get install unzip)

VERSION=`cat currentversion`

if [ -d build ]; then
     echo "Start building engengis..."
     sleep 1
else
     mkdir build
fi;
cd include
echo "Making zip archieve..."
sleep 1
zip -r engengis * 
cd ..
sleep 2
echo "Signing zip archieve..."
mv include/engengis.zip signzip/engengis.zip
cd signzip
java -jar signapk.jar testkey.x509.pem testkey.pk8 engengis.zip signed-engengis.zip
cd ..
mv signzip/signed-engengis.zip build/$VERSION.zip
echo "Done find build at:"
echo "/build/$VERSION.zip"
rm -f signzip/engengis.zip
sleep 3


updater.sh
#!/system/bin/sh
# Copyright© 2011 redmaner
# Engengis updater V16

checkupdate () {
clear
CONFIG=/data/engengis.conf
VERSION=46
OLDVERSION=`cat /system/etc/engengis/version`

echo
echo " ------------------------"
echo "    Engengis.Updater   " 
echo " ------------------------"
echo
echo "Latest version = Engengis.Charlie_$VERSION"
echo
if [ $OLDVERSION -eq $VERSION ]; then
     echo "Update is not needed"
     echo "Do you want to force an update?"
     echo "[y or n]"
     read forceupdate
elif [ $OLDVERSION -lt $VERSION ]; then
     echo "There is an update available"
     echo "Do you want to update?"
     echo "[y or n]"
     read update
elif [ $OLDVERSION -gt $VERSION ]; then
     echo "You are using a newer version off Engengis"
     echo "Your version is not yet supported by the mainline"
     echo "Update cancelled"
     sleep 3
     clear
     engengis 
fi;

case "$update" in
  "y" | "Y")
  mount -o remount,rw /system
  echo
  echo "This will take a moment..."
  mount -o remount,rw /system
  rm -rf /system/etc/engengis
  rm -f /system/bin/engengis
  rm -f /system/bin/engengisa
  rm -f /system/bin/engengissettings
  rm -f /system/bin/engengisg
  rm -f /system/bin/engengishss
  rm -f /system/bin/engengisr
  rm -f /system/bin/engengiss
  rm -f /system/xbin/sqlite3
  rm -f /system/xbin/zipalign
  rm -f /system/etc/init.d/S00ramscript
  rm -f /system/etc/init.d/S07hsstweak
  rm -f /system/etc/init.d/S14zipalign
  rm -f /system/etc/init.d/S21sqlite
  rm -f /system/etc/init.d/S28scheduler
  rm -f /system/etc/init.d/S35sdreadspeed
  rm -f /system/etc/init.d/S42cpugovernor
  rm -f /system/etc/init.d/S49dropcaches
  rm -f /system/etc/init.d/S56internet
  rm -f /system/etc/init.d/S63internetsecurity
  rm -f /data/engengis.log
  echo "Installing new files..."
  mkdir /system/etc/engengis
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/bin/engengis -O /system/bin/engengis
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S07hsstweak -O /system/etc/engengis/S00ramscript
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S14zipalign -O /system/etc/engengis/S14zipalign
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S21hsstweak -O /system/etc/engengis/S21hsstweak
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28bfqscheduler -O /system/etc/engengis/S28bfqscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28cfqscheduler -O /system/etc/engengis/S28cfqscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28deadlinescheduler -O /system/etc/engengis/S28deadlinescheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28noopscheduler -O /system/etc/engengis/S28noopscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28sioscheduler -O /system/etc/engengis/S28sioscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S28vrscheduler -O /system/etc/engengis/S28vrscheduler
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd256 -O /system/etc/engengis/S35sd256
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd512 -O /system/etc/engengis/S35sd512
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd1024 -O /system/etc/engengis/S35sd1024
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd2048 -O /system/etc/engengis/S35sd2048
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd3072 -O /system/etc/engengis/S35sd3072
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S35sd4096 -O /system/etc/engengis/S35sd4096
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpuconservative -O /system/etc/engengis/S42cpuconservative
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpululzactive -O /system/etc/engengis/S42cpululzactive
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpuinteractive -O /system/etc/engengis/S42cpuinteractive
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpuondemand -O /system/etc/engengis/S42cpuondemand
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S42cpusmartassv2 -O /system/etc/engengis/S42cpusmartassv2
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S49dropcaches -O /system/etc/engengis/S49dropcaches
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S56internet -O /system/etc/engengis/S56internet
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/S63internetsecurity -O /system/etc/engengis/S63internetsecurity
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/etc/engengis/version -O /system/etc/engengis/version
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/xbin/sqlite3 -O /system/xbin/sqlite3
  wget -q http://dl.dropbox.com/u/26139869/engengis/system/xbin/zipalign -O /system/xbin/zipalign
  echo "Setting permissions..."
  chmod 777 /system/bin/engengis
  chmod 777 /system/xbin/sqlite3
  chmod 777 /system/xbin/zipalign
  if [ -e /data/settings.conf ]; then
      sed -i '/status=*/ d' /data/engengis.conf;
      echo "status=updated" >> /data/engengis.conf;
  fi;
  echo
  echo "Installation complete!"
  sleep 5
  clear 
  engengis
  ;;
  "n" | "N")
  echo "Update cancelled"
  sleep 3
  clear
  engengis 
  ;;
esac

case "$forceupdate" in
  "y" | "Y")
  echo "1" > $OLDVERSION
  clear; checkupdate;;
  "n" | "N") clear; engengis;;
esac
}

checkupdate;



